### PR TITLE
🚀  Add support for Icon color and size to MenuItem component

### PR DESCRIPTION
**Pull request by Jejugo**
[JIRA Issue](https://gympass.atlassian.net/browse/)

## Description 📄

The main idea of this PR is to be able to add color and size to an Icon inside the Menu Item component. As of now we have the default values set to:

```
colors: #9898A6
size: 20px
```

## Platforms 📲

- [x] Web
- [ ] Mobile

## Type of change 🔍

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? 🧪

[Enter the tips to test this PR]

- [x] Unit Test
- [ ] Snapshot Test

## Checklist: 🔍

- [x] My code follows the contribution guide of this project [Contributing Guide](https://github.com/Gympass/yoga/blob/master/CONTRIBUTING.md)
- [x] Layout matches design prototype: [FIGMA](https://figma.com/file/YOUR_LINK)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Screenshots 📸

[Upload your screenshots here]

Before:
image

```
<Box display="flex" flexDirection="column" alignItems="center">
  <Menu onHover>
    <Menu.Action>
      <Button.Icon icon={Home} />
    </Menu.Action>

    <Menu.List align="end">
      <Menu.Item icon={Refresh} fill="#D8385E"><Text>Olá</Text></Menu.Item>
    </Menu.List>
  </Menu>
</Box>
```
After:
image

```
<Box display="flex" flexDirection="column" alignItems="center">
  <Menu onHover>
    <Menu.Action>
      <Button.Icon icon={Home} />
    </Menu.Action>

    <Menu.List align="end">
      <Menu.Item icon={Refresh} iconColor="#D8385E" iconSize="large"><Text>Olá</Text></Menu.Item>
    </Menu.List>
  </Menu>
</Box>
```

### CODE REVIEW

**Review by diegomarcuz — APPROVED** [review]

File: packages/yoga/src/Menu/web/MenuItem.jsx [hunk 1]
```diff
@@ -83,15 +98,18 @@ const MenuItem = forwardRef(
       finalProps.href = href;
     }
 
+    const size = get(theme.yoga.spacing, iconSize, iconSize) || 20;
```
Comment (MicaelRodrigues) [inline packages/yoga/src/Menu/web/MenuItem.jsx:101]: Shouldn't `20` be the default value?
```suggestion
const size = get(theme.yoga.spacing, iconSize, 20);
```

File: packages/yoga/src/Menu/web/MenuItem.jsx [hunk 2]
```diff
@@ -50,7 +52,7 @@ const StyledMenuItem = styled.li`
 
     svg {
       margin-right: ${menu.margin.default}px;
-      fill: ${menu.icon.disabled};
+      fill: ${iconColor};
```
Comment (diegomarcuz) [inline packages/yoga/src/Menu/web/MenuItem.jsx:55]: I was thinking here and would be good to set the yoga token as the colors, the prop keeps, but using the same idea as the `size`.

**Review by guilhermecardoso-gympass — APPROVED** [review]

**Review by guilhermecardoso-gympass — CHANGES_REQUESTED** [review]
remove undefined in defaultProps in MenuItem!

**Comment by Jejugo** [comment]
As discussed with @guilhermecardoso-gympass we decided to keep the `defaultProps` and keep setting the items to undefined to keep the code standard as it is today. This issue will be addressed in the future in a later commit

**Comment by Jejugo** [comment]
> remove undefined in defaultProps in MenuItem!

As discussed with @guilhermecardoso-gympass we decided to keep the defaultProps and keep setting the items to undefined to keep the code standard as it is today. This issue will be addressed in the future in a later commit

**Review by MicaelRodrigues — APPROVED** [review]
I'll approve this. But don't agree to keep bad code just because other bad code was already in the project.

**Review by guilhermecardoso-gympass — APPROVED** [review]

**Pull request merged**